core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -181,6 +181,9 @@ void *CudaSpace::allocate(const size_t arg_alloc_size) const {
 
   auto error_code = cudaMalloc(&ptr, arg_alloc_size);
   if (error_code != cudaSuccess) {  // TODO tag as unlikely branch
+    cudaGetLastError(); // This is the only way to clear the last error, which
```

Comment:
You may want to cast the return value to void, to avoid an "unused return value" warning.